### PR TITLE
Improvements to CSS grid layout.

### PR DESCRIPTION
**Pull request by toddburnside**
Also allows the flex container to be used outside of a grid.

### CODE REVIEW

File: common/src/main/resources/less/gpp-grid.less [hunk 1]
```diff
@@ -1,6 +1,15 @@
 @two-column-break: 60em;
 @three-column-break: 90em;
 
+/*
+ Overrides the default values of fields inside an SUI form.
+ These values interfere with the gpp-grid layout.
+ */
+.ui.form.gpp-grid .field {
+    margin: 0;
```
Comment (toddburnside) [inline common/src/main/resources/less/gpp-grid.less:9]: I was following prior art, but should we change all the `gpp` to `explore` or `lucuma`, and rename `GPPStyles` similarly?

Comment (cquiroz) [reply common/src/main/resources/less/gpp-grid.less:9]: I suppose we should, maybe explore considering it is explore specific but if it is going to be used more broadly, lucuma?

File: common/src/main/resources/less/gpp-grid.less [hunk 2]
```diff
@@ -31,61 +40,42 @@
...
+ nature of the grid - they will remain grouped together even
+ when the grid goes down to one column.
+*/
+.gpp-flex-container {
+    display: flex;
 
-            &-three {
-                flex: 3;
-            }
+    .ui.form & > * + *, // override SUI
```
Comment (cquiroz) [inline common/src/main/resources/less/gpp-grid.less:57]: Cool, I didn't know this construct

File: common/src/main/scala/explore/components/ui/GPPStyles.scala [hunk 3]
```diff
@@ -71,15 +73,12 @@ object GPPStyles {
...
   val TwoColumnGrid: Css   = Grid |+| Css("gpp-two-columns")
   val ThreeColumnGrid: Css = Grid |+| Css("gpp-three-columns")
-  val CellWrapper: Css     = Css("gpp-cell-wrapper")
 
-  val GrowOne: Css   = Css("gpp-grow-one")
-  val GrowTwo: Css   = Css("gpp-grow-two")
-  val GrowThree: Css = Css("gpp-grow-three")
-  val GrowFour: Css  = Css("gpp-grow-four")
+  val FlexContainer: Css = Css("gpp-flex-container")
+  def Grow(i: Int Refined Interval.Closed[1, 4]): Css = Css(s"gpp-grow-$i")
```
Comment (cquiroz) [inline common/src/main/scala/explore/components/ui/GPPStyles.scala:78]: Neat

**Review by cquiroz — APPROVED** [review]
This is looking great, some good use of less constructs

File: common/src/main/resources/less/gpp-grid.less (hunk 2 above)
Comment (rpiaggio) [reply common/src/main/resources/less/gpp-grid.less]: 🤯

**Review by rpiaggio — APPROVED** [review]
It's very nice to have these unified layout styles. LGTM!

**Pull request merged**